File: 2022/01/day1.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"

	"AoC/utils"
)

func main() {
	input := utils.CheckErr(os.ReadFile("input.txt"))

	elfCalorieLists := [][]int{{}}
	for _, line := range utils.Lines(string(input)) {
		if line != "" {
			val := utils.CheckErr(strconv.Atoi(line))
			currentList := &elfCalorieLists[len(elfCalorieLists)-1]
			*currentList = append(*currentList, val)
		} else {
			elfCalorieLists = append(elfCalorieLists, []int{})
		}
	}

	elfCalorieCounts := utils.Transform(elfCalorieLists, utils.Sum)
	sort.Ints(elfCalorieCounts)

	size := len(elfCalorieCounts)
	fmt.Printf("The answer to Part 1 is %v.\n", elfCalorieCounts[size-1])
	fmt.Printf("The answer to Part 2 is %v.\n", utils.Sum(elfCalorieCounts[size-3:]))
}

File: 2015/01/day1.py
with open("input.txt") as f:
    input = f.readline().strip()

floor = 0
enter_basement_pos = None
for (i, c) in enumerate(input):
    floor = floor + 1 if c == "(" else floor - 1
    if enter_basement_pos is None and floor == -1:
        enter_basement_pos = i + 1

print(f"The answer to Part 1 is {floor}.")
print(f"The answer to Part 2 is {enter_basement_pos}.")
